source/nodejs/adaptivecards/src/serialization.ts

```diff
@@ -1086,10 +1086,23 @@ export abstract class SerializableObject {
         }
     }
 
-    protected getValue(prop: PropertyDefinition): any {
-        return this._propertyBag.hasOwnProperty(prop.getInternalName())
-            ? this._propertyBag[prop.getInternalName()]
-            : prop.defaultValue;
+    protected getValue(prop: PropertyDefinition, overrideProperty: boolean = false): any {
+        const name = prop.getInternalName();
+        if (this._propertyBag.hasOwnProperty(name)) {
+            const currentValue = this._propertyBag[name];
+
+            if (overrideProperty && this._rawProperties.hasOwnProperty(name)) {
+                const overrideValue = this._rawProperties[name];
+
+                if (currentValue !== overrideValue && typeof overrideValue === "string" && overrideValue.includes("${")) {
```

Comment:
i don't like using the template token, "${" directly. the token can be changed and more than one token.
also if we allow override, i don't think we really need to check if it's template or not.

Reply:
I think this should be fine. My only concern is for properties where `propertyBag` and `rawProperties` hold different values and there's not templating (ex: enums).

I'm not seeing any issues, but it seems like this could cause some unforeseen errors. Any thoughts?

Reply:
that should be fine. your change is applied during serialization, and we want full round trip.
during deserialization, bad values will get good default values.
you could consider

if (shouldOverride)
elif (has PropertyBagValue)

return property default.